MaxText/layers/llama2.py

```diff
@@ -62,6 +62,7 @@ def __call__(self,
                inputs,
                decoder_segment_ids,
                decoder_positions,
+               padding_mask,
```

Comment:
Can you just move all the Decoder Layer new arguments (i.e. other than inputs, decoder_segment_ids and decoder_positions to be keyword to simplify this going forwards?

Reply:
I have a question whether we have limitations in modifying arguments and keywords from shared`nn.scan` api inside `Decoder` class among different models. I think it requires us to ensure the exact same decoder_layer arguments among different models. I am really interested in how to walk around the limitation to allow different arguments for different decoder_layer. Then, we can hide unused arguments in `**kwargs`. 
```
class Decoder(nn.Module):
...
     BlockLayer = nn.remat(  # pylint: disable=invalid-name
         ...
          static_argnums=(-1, -2, -3, -4, -5, -6),
      )

        y, _ = nn.scan(
          BlockLayer,
...
          in_axes=(
              nn.broadcast,
              nn.broadcast,
              nn.broadcast,
              nn.broadcast,
              nn.broadcast,
          ),
....
      )(config=cfg, mesh=mesh, name='layers')(
          y,
          decoder_segment_ids,
          decoder_positions,
          padding_mask,
          deterministic,
          model_mode,
      )
```

Reply:
This is a conversation we can have but can you start by explaining why we want this padding to be passed separately? The currrent design makes sense if we don't need to make mnay changes here -- if we do, we need to have a conversation...

Reply:
Discussed offline. Remove `padding_mask`